### PR TITLE
Feature/job families

### PR DESCRIPTION
**Pull request by PatrickWatson887**

### CODE REVIEW

File: Doras/src/app/carousel-tabs/carousel-tabs.component.html [hunk 1]
```diff
@@ -1,9 +1,12 @@
 <ngb-tabset justify="fill">
-    <li *ngFor ="let capability of capabilities|async">
-        <ngb-tab title="{{capability.capName}}">
+  <ul>
+    <li *ngFor ="let capability of capabilities">
+      <p>{{capabilities}}</p>
```
Comment (micalexis) [inline Doras/src/app/carousel-tabs/carousel-tabs.component.html]: this should be removed

File: Doras/src/app/carousel-tabs/carousel-tabs.component.ts [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
   selector: 'doras-carousel-tabs',
   templateUrl: './carousel-tabs.component.html',
   styleUrls: ['./carousel-tabs.component.css'],
 })
-export class CarouselTabsComponent implements OnInit {
+export class CarouselTabsComponent implements OnInit, OnDestroy {
+  public capabilities: Capability[];
+
+  constructor(private switchBoard: SwitchBoardService) {
+    console.log(this.capabilities);
```
Comment (micalexis) [inline Doras/src/app/carousel-tabs/carousel-tabs.component.ts]: this should be removed

File: Doras/src/app/carousel/carousel.component.ts [hunk 3]
```diff
@@ -25,17 +28,30 @@ import { HttpClient } from '@angular/common/http';
...
     config.keyboard = false;
     config.pauseOnHover = false;
-
   }
-  capabilities = this.http.get<string[]>('/api/getCapabilities');
-  
 
-  ngOnInit() {
+  onSlide(slideEvent: NgbSlideEvent) {
+    let index = slideEvent.current.split('-')[2];
```
Comment (micalexis) [inline Doras/src/app/carousel/carousel.component.ts]: did we not discuss changing this from using the slide id to the actual DB id?

File: Doras/src/app/data.service.ts [hunk 4]
```diff
@@ -3,26 +3,37 @@ import { HttpClient } from '@angular/common/http'
...
 
-  public getCapabilities(): Promise<Capability[]> {
-    return this.http.get<Capability[]>('/api/getCapabilities').toPromise()
+  public getCapabilities(): Observable<Capability[]> {
+    return this.http.get<Capability[]>('/api/getCapabilities');
   }
 
-  public getJobFamilies(): Promise<JobFamily[]> {
-    return this.http.get<JobFamily[]>('/api/getJobFamilies').toPromise()
+  public getJobFamily(): Promise<JobFamily[]> {
```
Comment (micalexis) [inline Doras/src/app/data.service.ts]: follow one approach! refactor all endpoints to use Observables!

File: ExpressAPI/app.js [hunk 5]
```diff
@@ -41,6 +41,21 @@ app.get('/getJobFamily', function(req, res) {
...
 })
 
+function getCapNameByJfId(id, capNameReadyFn) {
+  db.getCapNameByJfid(id, function(rows) {
+      capName = rows
+      capNameReadyFn()
+  })
+}
+
+app.get('/getCapNameByJfId/:id', function(req, res) {
```
Comment (micalexis) [inline ExpressAPI/app.js:51]: your express endpoints need to be RESTful, look into this!

**Review by micalexis — CHANGES_REQUESTED** [review]
address the comments and squash your commits

File: Doras/src/app/roles/roles.component.html [hunk 6]
```diff
@@ -1,16 +1,51 @@
...
+        <ngb-carousel data-interval="false" class="carouselTop" (slide)="onSlide($event)">
+            <li *ngFor ="let band of endArray; let i = index">
+                <ng-template ngbSlide>
 
-</div>
\ No newline at end of file
+                    <table class="tableBand">
+                        <tr>
+                            <th 
+                     class="th1" scope="col">{{endArray[i][0].bandName}}
```
Comment (micalexis) [inline Doras/src/app/roles/roles.component.html]: what is endArray? give it a proper name

File: Doras/src/app/roles/roles.component.html [hunk 7]
```diff
@@ -1,16 +1,51 @@
...
+                                <h4 *ngIf="job1Title; else notShowBlock">{{job1Title}}</h4>
```
Comment (micalexis) [inline Doras/src/app/roles/roles.component.html:11]: job1Title, job2Title, job3Title? why are we not looping over these? this needs to be refactored!!

File: Doras/src/app/roles/roles.component.ts [hunk 8]
```diff
@@ -1,15 +1,86 @@
...
 @Component({
   selector: 'app-roles',
   templateUrl: './roles.component.html',
   styleUrls: ['./roles.component.css']
 })
 export class RolesComponent implements OnInit {
+  public bands: Band[];
+  public endArray: any[];
+  public count: number = 0;
+  public jobs1: Job[];
```
Comment (micalexis) [inline Doras/src/app/roles/roles.component.ts:18]: what do these refer to?

File: Doras/src/app/roles/roles.component.ts [hunk 9]
```diff
@@ -1,15 +1,86 @@
...
+    return this.dataTransferService.capability;
+}
+
+  constructor(private data: DataService, private dataTransferService : DataTransferService) { }
 
   ngOnInit() {
+    this.data.getBand().subscribe(c => {
+      this.bands = c.reverse();
+
+      let array = []
```
Comment (micalexis) [inline Doras/src/app/roles/roles.component.ts]: where are the `;` at the end of each statement?

**Review by lukemciKainos — APPROVED** [review]

**Review by micalexis — COMMENTED** [review]
make sure you squash your commits!!!!

**Pull request merged**